### PR TITLE
rs string functions

### PR DESCRIPTION
**Pull request by nichtsfrei**
Implement substr
    Implement string
    Implement strlen
    Implement tolower and toupper
    Implement raw_string
    Implement hexstr

### CODE REVIEW

File: rust/nasl-interpreter/src/built_in_functions/string.rs [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+                (&s[lidx as usize..ridx as usize]).into()
+            }
+            _ => (&s[lidx as usize..]).into(),
+        })
+    }
+}
+
+/// NASL function to return  a hex presentation of given string as a positional argument.
+///
+/// If the positional arguments are empty it reutnrns NaslValue::Null.
```
Comment (Kraemii) [inline rust/nasl-interpreter/src/built_in_functions/string.rs:161]:
```suggestion
/// If the positional arguments are empty it returns NaslValue::Null.
```

File: rust/nasl-interpreter/src/built_in_functions/string.rs [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+            Some(nv) => {
+                let ridx: i64 = nv.into();
+                (&s[lidx as usize..ridx as usize]).into()
+            }
+            _ => (&s[lidx as usize..]).into(),
+        })
+    }
+}
+
+/// NASL function to return  a hex presentation of given string as a positional argument.
```
Comment (Kraemii) [inline rust/nasl-interpreter/src/built_in_functions/string.rs:159]:
```suggestion
/// NASL function to return a hex representation of a given positional string argument.
```

File: rust/nasl-interpreter/src/built_in_functions/string.rs [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+///
+/// If the positional arguments are empty it reutnrns NaslValue::Null.
+/// It only uses the first positional argument and when it is not a NaslValue:String than it returns NaslValue::Null.
+fn hexstr(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let positional = resolve_positional_arguments(register);
+    // we already checked that positional has at least one argument
```
Comment (Kraemii) [inline rust/nasl-interpreter/src/built_in_functions/string.rs:165]: I am pretty sure this is not true, as resolve_positional_argument should return an empty vec, if no positional arguments were given. In the end we will return `NaslValue::NULL` in this case, as it is covered within the followed match block.

Comment (nichtsfrei) [reply rust/nasl-interpreter/src/built_in_functions/string.rs:165]: yes, that comment is a leftover of a previous solution. Should be deleted

**Review by Kraemii — APPROVED** [review]

**Review by ArnoStiefvater — APPROVED** [review]

**Pull request merged**